cmd/cmd.go

```diff
@@ -0,0 +1,58 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package cmd
+
+import (
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/ubuntu-core/snappy/osutil"
+	"github.com/ubuntu-core/snappy/release"
+)
+
+const key = "SNAP_INNER"
+
+// SwitchToTheRealOne makes sure you're executing the "real"
+// binary. I.e. the one that ships in the ubuntu-core snap.
+func SwitchToTheRealOne() {
```

Comment:
Can we please rename this to `ExecInCoreSnap`?  They are both very real, so it'll be easier to read if we drive some sort of distinction in the name.